doc/thingpedia-device-class.md

```diff
@@ -1,2 +1,113 @@
 # Writing Device Class in ThingTalk
 
+[[toc]]
+
+Device class describes how the device will be loaded by the system, how it will 
+be configured, and what it does. 
+In the following, we will continue using The Cat API as an example to illustrate 
+how to write your own device class. The device class of it looks like this:
+
+```tt
+class @com.thecatapi {
+  import loader from @org.thingpedia.v2();
+  import config from @org.thingpedia.config.none();
+
+  query get(in opt count: Number,
+            out image_id: Entity(com.thecatapi:image_id),
+            out picture_url: Entity(tt:picture),
+            out link: Entity(tt:url))
+  #_[canonical="get cat on thecatapi"]
+  #_[confirmation="cat pictures"]
+  #[doc="get `count` many cat pictures"];
+}
+```
+The name of a device class is as defined in the device metadata with an additional `@` 
+sign in front. The device class has two types of statements: import statements and function declaration statements.
+Each statement terminates with a semicolon. 
+
+## Import statements
+The import statements import modules `loader` and `config` for the device, 
+to specify how the device will be loaded by the system and how it will be initially configured 
+or authenticated. 
+
+Mixin `@org.thingpedia.v2()` provides the most flexible loader module, which allows 
+the user to supply additional JS package code to customized the configuration of the device 
+and the behavior of each function. 
```

Comment:
Is this the first time we introduce the concept of JS package?
Maybe the whole JS discussion should be moved away from the overview first page, and into its own page.
Logically, it should go: high level overview -> ThingTalk class ->  JS code.